circuit/src/queryBuilderBase/dataSubquery/format.ts

```diff
@@ -73,7 +73,7 @@ function formatDataSubqueryHeader(subquery: HeaderSubquery): DataSubquery {
     subqueryData: {
       blockNumber: subquery.blockNumber,
       fieldIdx: subquery.fieldIdx,
-    },
+    } as HeaderSubquery,
   };
```

Comment:
You're passing in HeaderSubquery but casting to HeaderSubquery again. What is going on here? Also typescript casting is unsafe (does do any checks), so it may be best to avoid.

Reply:
The main issue is that adding `blockNumber?` to `Subquery` in the axiom-tools PR causes some downstream issues with interface types, so they unfortunately need to be casted or the typescript compiler will not be happy

The unit tests in tools all had to be updated as such due to this change: https://github.com/axiom-crypto/axiom-tools/pull/3/files

Reply:
ah, so actually subqueryData is of type `Subquery`, of which `HeaderSubquery` is derived, and we're casting `subqueryData` within the return object, not the entire return object. 

However, I think you bring up a good point and i'll update it to throw an error if blockNumber is not included instead of trying to cast.

Reply:
Nevermind; we need to keep the cast to help tsc discern the derived `Subquery` interfaces